### DISCUSSION TITLE
ReValidation of hidden inputs

### DISCUSSION THREAD

**Discussion by adammo94** [react-hook-form/react-hook-form#7373] [Q&A]
I have a few custom components, f.e. datepicker, dropdown.
I want to revalidate its' values. However revalidation only goes off when user manually changes the input.
Is there a possiblity to trigger revalidation on value that was changed externally (let's say from buttons)? 

```
export const MyCustomField = ({name, register, error}) => {
  const [value, setValue] = useState('')
  
  <>
    <input type="hidden" value={value} ref={register} name={name}/>
    <button onClick={()=>setValue(Math.rand()) />
    <button onClick={()=>setValue('')/>
    {
      error && <p>{error}</p>
  </>
}
```

I tried with setValue, but it will validate even when form is not submitted.
Can I somehow check if value has changed after submitting and then do revalidation?

**Comment by bluebill1049** [comment]
1. https://react-hook-form.com/api/useform/trigger
2. https://react-hook-form.com/api/useform/setvalue (`{ shouldValidate: true }`)
